packages/react-composites/src/composites/MeetingComposite/state/MeetingState.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CallState } from '@internal/calling-stateful-client';
+import { ChatThreadClientState } from '@internal/chat-stateful-client';
+import { MeetingParticipant } from './MeetingParticipants';
+import { MeetingEndReason } from './MeetingEndReason';
+import { convertCallParticipantsToMeetingParticipants } from '../state/MeetingParticipants';
+
+export interface MeetingState
+  extends Pick<
+      CallState,
+      | 'callerInfo'
+      | 'state'
+      | 'isMuted'
+      | 'isScreenSharingOn'
+      | 'localVideoStreams'
+      | 'transcription'
+      | 'recording'
+      | 'transfer'
+      | 'screenShareRemoteParticipant'
+      | 'startTime'
+      | 'endTime'
+    >,
+    Pick<
+      ChatThreadClientState,
+      'chatMessages' | 'threadId' | 'properties' | 'readReceipts' | 'typingIndicators' | 'latestReadTime'
+    > {
+  id: string;
+  participants: { [key: string]: MeetingParticipant };
+  participantsEnded: { [keys: string]: MeetingParticipant };
+  meetingEndReason?: MeetingEndReason;
+}
+
+export function generateMeetingState(callState: CallState, chatState: ChatThreadClientState): MeetingState {
+  const {
+    callerInfo,
+    state,
+    isMuted,
+    isScreenSharingOn,
+    localVideoStreams,
+    transcription,
+    recording,
+    transfer,
+    screenShareRemoteParticipant,
+    startTime,
+    endTime,
+    id,
+    remoteParticipants,
+    remoteParticipantsEnded,
+    callEndReason
+  } = callState;
+
+  // Only use participants from the call in a meeting
+  const participants = remoteParticipants;
+  const participantsEnded = remoteParticipantsEnded;
+
+  const { chatMessages, threadId, properties, readReceipts, typingIndicators, latestReadTime } = chatState;
```

Comment:
tbh, this function will be shorter and more readable if you don't use destructuring:

    return {
      id: callState.id,
      ...
      threadId: chatState.threadId,
      ...
    }

Reply:
The reason I used the destructing was to be explicit on what it being set.
For example if someone adds a new field to both calling and chat states e.g. `ownerID: string`
If we use destructing this will be satisfied without having been updated when in reality someone will need to decide which `owner` to use.

But as I'm typing this its exactly why I need unit tests and they will solve this problem.

I'll update this to use destructing 😊

Reply:
That simplified a lot of functions thanks!